### ISSUE TITLE
I alway have a problem with Node-Gyp when npm install

### ISSUE THREAD

**Issue by artcheeze** [D-Andreev/recommender-addon#26]
I'm using recommender v3.0.0 and i dev on Window with Working pretty good!
but when i take it to production server (Debian) and npm install 
i alway get install error with node-gyp 

* Solution that i try but not work
- Delete .node_module
- Delete Package-lock.json
- Delete node-gyp folder

![image](https://user-images.githubusercontent.com/25684433/65596652-7f5d1680-dfc1-11e9-977d-99a08620b13a.png)

![image](https://user-images.githubusercontent.com/25684433/65596662-86842480-dfc1-11e9-9d12-269a4127e2d7.png)

I don't know how to solve this problem.

**Comment by D-Andreev** [comment]
Yes, I remember I had similar (or maybe the same) issue when running my tests on travis ci. There it's Debian as well. node-gyp requires some dependencies which are not installed by default on Debian. 

Can you try installing the dependencies from `before_install` section here - https://github.com/D-Andreev/recommender-addon/blob/master/.travis.yml#L5 on your production env?

**Comment by romayalon** [comment]
Hey @artcheeze @D-Andreev ,
I'm trying to install the library on windows but having some issues with node-gyp.
At the beginning it was an windows 8.1 SDK error and after installing it I get the following error: 
TRACKER : error TRK0005: Failed to locate: "CL.exe".
is this repo is still maintained? 

@artcheeze have you experienced these kind of errors on windows? or you just ran npm install recommender and it worked?

Thanks

**Comment by artcheeze** [comment]
> Hey @artcheeze @D-Andreev ,
> I'm trying to install the library on windows but having some issues with node-gyp.
> At the beginning it was an windows 8.1 SDK error and after installing it I get the following error:
> TRACKER : error TRK0005: Failed to locate: "CL.exe".
> is this repo is still maintained?
> 
> @artcheeze have you experienced these kind of errors on windows? or you just ran npm install recommender and it worked?
> 
> Thanks

My infrastructure team told me 
" i install build chain (or something that he cannot remember ) on Ubuntu Server before run npm install again and then it's work "

Now, i got contactless with him that's it everyting i known

**Comment by thisalihassan** [comment]
Hi @artcheeze  @artcheeze  
I am on windows and getting this error while installing recommender npm npm ERR! recommender@3.0.0 install: `node-gyp rebuild`
I have looked everywhere for a solution but nothing seems to be working for me
![image](https://user-images.githubusercontent.com/24819103/83956486-2c3da200-a878-11ea-8458-b15b7601f51e.png)

**Comment by D-Andreev** [comment]
@thisalihassan @romayalon 

For Windows I would try installing https://www.npmjs.com/package/windows-build-tools first. What node versions are you running? This package was published when stable version of node was 7.0.0, so things could have changed.

**Comment by thisalihassan** [comment]
@D-Andreev 
I am using node 12.6.0. I have already installed windows build tools
here's the log fle that you asked for
[log.txt](https://github.com/D-Andreev/recommender-addon/files/4741546/log.txt)

**Comment by D-Andreev** [comment]
@thisalihassan I just tried it on my Windows machine with node 12 and could not install it. By switching node to version `8.0.0` i was able to install the package. Let me know if this helps.

**Comment by thisalihassan** [comment]
@D-Andreev 
Thank you for the help, it works on node 8

**Comment by onepiecehung** [comment]
> @thisalihassan I just tried it on my Windows machine with node 12 and could not install it. By switching node to version `8.0.0` i was able to install the package. Let me know if this helps.

@D-Andreev
How can I use this package on node v12 LTS on windows? 

![image](https://user-images.githubusercontent.com/24888125/86263631-af3de800-bbeb-11ea-8007-14bb12726a1d.png)

**Comment by thisalihassan** [comment]
@onepiecehung You can't actually as, this package uses node 8.0.0 or less. There's nothing we could do to run on node 12

**Comment by thisalihassan** [comment]
@onepiecehung I can recommend you an alternative to this. I used "content-based-recommender" I hope this could help you

**Comment by onepiecehung** [comment]
> @thisalihassan 

oh, I can install and run it in node v10, on google shell and codesandbox.io , but it's still not working in my PC. lol

**Comment by thisalihassan** [comment]
what's the error? did you try [ https://www.npmjs.com/package/windows-build-tools](url) installing this?

**Comment by onepiecehung** [comment]
> what's the error? did you try [ https://www.npmjs.com/package/windows-build-tools](url) installing this?

I have installed these tools before installing this npm
